third_party/nvfuser/csrc/ir_nodes.cpp

```diff
@@ -2996,4 +3074,209 @@ c10::optional<ParallelType> NamedScalar::getParallelIndex() const {
   return c10::nullopt;
 }
 
+PadOp::PadOp(
+    IrBuilderPasskey passkey,
+    TensorView* out,
+    TensorView* inp,
+    const std::vector<Val*>& pad_widths)
+    : Expr(passkey) {
+  addOutput(out);
+  addInput(inp);
+  for (auto width : pad_widths) {
+    addAttribute(width);
+  }
+}
+
+NVFUSER_DEFINE_CLONE_AND_CREATE(PadOp)
+
+std::string PadOp::toString(int indent_size) const {
+  std::stringstream ss;
+  indent(ss, indent_size) << out()->toString() << "\n";
+  indent(ss, indent_size) << "   = pad( " << in()->toString() << ", {"
+                          << toDelimitedString(getPadWidths()) << "}"
+                          << " )\n";
+  return ss.str();
+}
+
+std::string PadOp::toInlineString(int indent_size) const {
+  TORCH_CHECK(false, "Tensor op can not be printed inline");
+}
+
+std::vector<int> PadOp::getPaddedAxes() const {
+  auto num_dims = out()->as<TensorView>()->getRootDomain().size();
+  auto num_padded_axes = attributes_.size() / 2;
+  std::vector<int> padded_axes(num_padded_axes);
+  std::iota(padded_axes.begin(), padded_axes.end(), num_dims - num_padded_axes);
+  return padded_axes;
+}
+
+std::vector<Val*> PadOp::getPadWidths() const {
+  std::vector<Val*> pad_widths;
+  std::transform(
+      attributes_.begin(),
+      attributes_.end(),
+      std::back_inserter(pad_widths),
+      [](Statement* attr) { return attr->as<Val>(); });
+  return pad_widths;
+}
+
+std::pair<Val*, Val*> PadOp::getPadWidths(int axis) const {
+  auto num_dims = static_cast<int>(out()->as<TensorView>()->nDims());
+  auto num_padded_axes = static_cast<int>(attributes_.size() / 2);
+
+  if (axis < 0) {
+    axis += num_dims;
+  }
+
+  TORCH_CHECK(axis >= 0 && axis < num_dims, "Invalid axis: ", axis);
+
+  // Just return zero for non padded domains
+  if (axis < num_dims - num_padded_axes) {
+    return std::make_pair(container()->zeroVal(), container()->zeroVal());
+  }
+
+  auto pad_idx = axis - (num_dims - num_padded_axes);
+  TORCH_INTERNAL_ASSERT(pad_idx >= 0 && pad_idx < num_dims);
+
+  return std::make_pair(
+      attribute(pad_idx)->as<Val>(), attribute(pad_idx + 1)->as<Val>());
```

Comment:
Why do I think this should be `2 * pad_idx` and `2 * pad_idx + 1`. Do I understand the semantics wrong?